Add locale limitation to python support doc (#1130)
deislabs/mystikos: doc/PythonSupport.md

 - `threading.Lock` - unsupported (yet)
- `curses` - not supported
- `tkinter` - not supported

> - `locale` - musl libc [has very limited locale support](https://wiki.musl-libc.org/open-issues.html). Python program running in Mystikos would suffer the same limitation, since Mystikos is using musl.

Reviewer: ```suggestion
- `locale` - musl libc [has very limited locale support](https://wiki.musl-libc.org/open-issues.html). Python programs running in Mystikos would suffer the same limitations, since Mystikos is using musl.
```

> - `locale` - musl libc [has very limited locale support](https://wiki.musl-libc.org/open-issues.html). Python program running in Mystikos would suffer the same limitation, since Mystikos is using musl.

Reviewer 2: The more accurate way to put it: `Mystikos bases its C-runtime on musl`

>   * The exact missing functionality related to locate in Python is not deteremined, however you can refer to failed Python unit test for a rough understanding:

Reviewer: ```suggestion
  * The exact missing functionality related to locate in Python is not deteremined, however you can refer to the failed Python unit tests for a rough understanding.
```

>   * For example, musl doesn't support `LC_COLLATE`, failing following Python unit tests

Reviewer: ```suggestion
  * For example, musl doesn't support `LC_COLLATE`, failing the following Python unit tests:
```

    * [`test_strcoll_with_diacritic, test_strxfrm_with_diacritic`](https://github.com/python/cpython/blob/f4c03484da59049eb62a9bf7777b963e2267d187/Lib/test/test_locale.py#L374-L382)

>   * Other failed unit tests related to locale

Reviewer: ```suggestion
  * Other failed unit tests related to locale:
```

- `curses` - not supported
- `tkinter` - not supported

**Caveat**:

* Any third-party Python library that calls into the unsupported
standard Python APIs are not supported neither. Sometimes they can produce
unpredictable errors which are tricky to root cause.

> - Locale support - musl libc [has very limited locale support](https://wiki.musl-libc.org/open-issues.html). Python program running in Mystikos would suffer the same limitation, since Mystikos is using musl.

Reviewer: Can you please give an example of how does the limited locale support affect a Python program? For example, as a Python developer, I might not know about MUSL or locale. What I care about is what kind of Python library or API calls would fail in Mystikos because of locale support.
